[MRG + 1] Return correct ridge parameter alpha_ and lambda_ for Bayesian ridge regression

gedeck opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#Contributing-Pull-Requests
-->
#### Reference Issue
Fixes #8224


#### What does this implement/fix? Explain your changes.
In the original implementation, the `alpha_` and `lambda_` were modified after calculation of `coef_` and were therefore slightly wrong. The change moves the assignment to the object attributes `self.alpha_` and `self.lambda_` to right after the calculation of `coef_`. 

#### Any other comments?
The test uses a comparison to the standard ridge regression. A ridge regression with a ridge parameter of `self.lambda_ / self.alpha_` must return an identical regression. In the test, we compare the calculated intercept of the Bayesian ridge regression with the standard ridge regression. Before the change, the intercepts were slightly different. 
```
    # ACTUAL: 2.422446078010811
    # DESIRED: 2.4224997161532529
```

gedeck committed:
Return correct ridge parameter alpha_ and lambda_ for regression

agramfort commented on sklearn/linear_model/tests/test_bayes.py:
```diff
@@ -34,6 +34,21 @@ def test_bayesian_on_diabetes():
+
+    # A Ridge regression model using an alpha value equal to the ratio of lambda_ and alpha_ from 
+    # the Bayesian Ridge model must be identical
+    brModel = BayesianRidge(compute_score=True).fit(X, y)
```
we don't use CamelCase for var names.

brModel -> br or br_model

same for rrModel

agramfort commented on sklearn/linear_model/tests/test_bayes.py:
```diff
@@ -34,6 +34,21 @@ def test_bayesian_on_diabetes():
+    # the Bayesian Ridge model must be identical
+    brModel = BayesianRidge(compute_score=True).fit(X, y)
+    rrModel = Ridge(alpha=brModel.lambda_ / brModel.alpha_).fit(X, y)
+    assert_almost_equal(rrModel.intercept_, brModel.intercept_)
```
can you also test the coef_ ?

gedeck committed:
Add test for coefficients and fix style

gedeck commented:
Implemented reviewer's suggestions:

- Removed use of camel case for variable names.
- Add tests for coefficients. They were also slightly different in the original implementation
```
 x: array([ 0.000774,  0.000859])
 y: array([ 0.000767,  0.000852])
```

gedeck committed:
Move sklearn.utils.testing to a more reasonable position.

gedeck committed:
Make flake8 happy

raghavrv commented on sklearn/linear_model/tests/test_bayes.py:
```diff
@@ -34,6 +35,22 @@ def test_bayesian_on_diabetes():
+    rr_model = Ridge(alpha=br_model.lambda_ / br_model.alpha_).fit(X, y)
+    assert_array_almost_equal(rr_model.coef_, br_model.coef_)
+    assert_almost_equal(rr_model.intercept_, br_model.intercept_)
+    # Results before fix
```
Please remove this...

raghavrv reviewed (changes requested):
+1 for merge pending comment and approval from @agramfort

raghavrv commented:
And a whatsnew entry...

agramfort commented:
after what's new entry and cleanup +1 for MRG

gedeck committed:
Code cleanup and entry in whats_new.rst

gedeck commented:
Addressed requested changes:

- Comment in `test_bayes.py` removed
- Entry added to `whats_new.rst`

raghavrv reviewed (approved):
Thanks for addressing the review!

agramfort commented:
thx @gedeck
